data/templates/centos-ks

```diff
@@ -154,6 +154,7 @@ export PATH
         sed -i '/^BOOTPROTO=/d' /etc/sysconfig/network-scripts/ifcfg-<%=n.device%>
         sed -i '/^ONBOOT=/d' /etc/sysconfig/network-scripts/ifcfg-<%=n.device%>
         echo "DEVICE=<%=n.device%>" >> /etc/sysconfig/network-scripts/ifcfg-<%=n.device%>
+        echo "HWADDR=<%=n.hwaddr%>" >> /etc/sysconfig/network-scripts/ifcfg-<%=n.device%>
```

Comment:
I think copying the ESXi design will be better, ESXi's `networkDevices[].device` can be either macAddress or interface name, the CentOS can also have such design. If it detects the device is MAC, then auto lookup its interface name and set the correct IP address against the interface name.